src/server.c

```diff
@@ -1173,25 +1173,12 @@ static void clientWritePauseDuringOOM() {
      * Pause also evictions, since we might reach below maxmemory after lazyfree */
     if ((getMaxmemoryState(NULL,NULL,NULL,NULL) == C_ERR) &&
         (bioPendingJobsOfType(BIO_LAZY_FREE))) {
-        if (!server.is_pause_write_clients_oom) {
             pauseActions(PAUSE_OOM_THROTTLE, LLONG_MAX,
                          PAUSE_ACTION_EVICT | PAUSE_ACTION_CLIENT_DENYOOM);
-            
-            elapsedStart(&server.stat_current_client_paused_oom_time);
-            server.is_pause_write_clients_oom = 1;
-        }
     } else {
-        if (server.is_pause_write_clients_oom) {
-            /* If not OOM && lazy-free, then unpause clients */
-            unpauseActions(PAUSE_OOM_THROTTLE);
-            if (server.stat_current_client_paused_oom_time) {
-                /* update statistics & reset current_client_paused_oom_time  */
-                server.stat_total_client_paused_oom_time += elapsedUs(
-                        server.stat_current_client_paused_oom_time);
-                server.stat_current_client_paused_oom_time = 0;
-            }
-            server.is_pause_write_clients_oom = 0;
-        }
+        /* If not OOM && lazy-free, then unpause clients */
+        long long elapsed_msec = unpauseActions(PAUSE_OOM_THROTTLE);
+        server.stat_total_client_paused_oom_time += elapsed_msec;
```

Comment:
i don't think it's a good idea to sum milliseconds rather than microseconds.
it's ok to report in msec, but the summing should be in usec so we don't lose precision

Reply:
in my mind was the issue that it is only being used for statistics at most. I will change it.